samza-kafka/src/test/java/org/apache/samza/system/kafka/MockKafkaProducer.java

```diff
@@ -232,7 +242,7 @@ public boolean isDone() {
 
     public FutureSuccess(ProducerRecord record, int offset) {
       this.record = record;
-      this._metadata = new RecordMetadata(new TopicPartition(record.topic(), record.partition() == null ? 0 : record.partition()), 0, offset, Record.NO_TIMESTAMP, -1, -1, -1);
+      this._metadata = new RecordMetadata(new TopicPartition(record.topic(), record.partition() == null ? 0 : record.partition()), 0, offset, -1L, -1, -1, -1);
```

Comment:
The class name changed. I thought it was OK to just use -1 since this is test code. My bad. I have updated the code to use `RecordBatch.NO_TIMESTAMP` instead.